### PR TITLE
fix(site): fix template schedule update overriding other settings

### LINKED ISSUE #13111: bug: auto update template setting overidden when updating template schedule
reported by a customer and I reproduced on `2.10.1`

> When I change a Schedule setting then under General settings the checkbox "Require workspaces automatically update when started" will be unchecked.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Closes https://github.com/coder/coder/issues/13111

### Minor design update

**Before**
Screenshot 2024-04-30 at 16 37 49

**After**
Screenshot 2024-04-30 at 16 37 33

### CODE REVIEW

File: codersdk/templates.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ type UpdateTemplateMeta struct {
 	// RequireActiveVersion mandates workspaces built using this template
 	// use the active version of the template. This option has no
 	// effect on template admins.
-	RequireActiveVersion bool `json:"require_active_version"`
+	RequireActiveVersion bool `json:"require_active_version,omitempty"`
```
Comment (BrunoQuaresma) [inline codersdk/templates.go:235]: I tested this a few times, including the e2e test, and it looks like it works ok by just adding `omitempty`.

Comment (sreya) [reply codersdk/templates.go:235]: Yeah it seems setting `omitempty` generates an optional type for the typescript payload. So you can omit the value in the scheduling form and it seems to set it to its existing value from the template.

**Review by sreya — APPROVED** [review]
LGTM thanks for fixing this

**Review by Kira-Pilot — APPROVED** [review]
nice!

**Pull request merged**